### ISSUE TITLE
Issue with tonic + prost (similar to example)

### ISSUE THREAD

**Issue by ddibiase** [hyperium/tonic#291]
### Version

tonic 0.1.1
prost 0.6
cargo 1.40.0
rust 1.41.1

Not possible to run cargo-tree (due to below reported error). Snippet of Cargo.toml below.

### Platform

Darwin me.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64

### Description

I have the following toml:

```toml
[dependencies]
prost-types = "0.6"
tonic = { version = "0.1.1", features = ["tls", "data-prost"] }
prost = "0.6"

[build-dependencies]
tonic-build = { version = "0.1.1", features = ["prost"] }
```

Here's my build.rs:

```rust
fn main() {
    tonic_build::prost::compile_protos("proto/helloworld.proto").unwrap();
    tonic_build::prost::compile_protos("proto/service.proto").unwrap();
}
```
Here are the errors I'm seeing when running cargo build or run:

```
error: failed to select a version for `tonic`.
    ... required by package `service v0.1.0 (/Users/ddibiase/Projects/service)`
versions that meet the requirements `= 0.1.1` are: 0.1.1

the package `service` depends on `tonic`, with features: `data-prost` but `tonic` does not have these features.


failed to select a version for `tonic` which could resolve this conflict
```
The above configurations are taken from the examples (mostly).

**Comment by ddibiase** [comment]
I've attempted checking out the master branch and building the examples, getting the following error:
```
error: failed to run custom build command for `tonic-examples v0.1.0 (/Users/ddibiase/Projects/tonic/examples)`

Caused by:
  process didn't exit successfully: `/Users/ddibiase/Projects/tonic/target/debug/build/tonic-examples-0ba1ffce65d0467d/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:1188:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed

**Comment by Perlmint** [comment]
Introducing new features `data-prost` of tonic and `prost` of tonic-build is not applied to the current release(0.1.1). IMO, You should check examples from https://github.com/hyperium/tonic/tree/v0.1.1.

**Comment by ddibiase** [comment]
@Perlmint did you make a mistake in referencing the version numbers? Both are referencing 0.1.1. Sorry I'm also slightly confused by the phrasing of your response. Mind clarifying?

**Comment by Perlmint** [comment]
@ddibiase No, both versions I referenced are same. I think that there is some misunderstanding. I'll try to clarify my previous response.

There are breaking changes - introducing `data-prost`, `prost` features you used - in master. And these changes are not included in v0.1.1.
So, If you want to use tonic v0.1.1 - latest release on crates.io, you should look up [examples from 0.1.1 tag](https://github.com/hyperium/tonic/tree/v0.1.1/examples), not from the master branch.

**Comment by ddibiase** [comment]
So I checked out that very branch and ran cargo build in the examples folder. Got the following log errors. I'm on Latest OS X.

```bash
cargo build                                                                                                                                               ✔ 
   Compiling proc-macro2 v1.0.9
   Compiling unicode-xid v0.2.0
   Compiling syn v1.0.16
   Compiling libc v0.2.67
   Compiling cfg-if v0.1.10
   Compiling log v0.4.8
   Compiling lazy_static v1.4.0
   Compiling bytes v0.5.4
   Compiling futures-core v0.3.4
   Compiling autocfg v1.0.0
   Compiling memchr v2.3.3
   Compiling slab v0.4.2
   Compiling fnv v1.0.6
   Compiling pin-project-lite v0.1.4
   Compiling tower-service v0.3.0
   Compiling getrandom v0.1.14
   Compiling futures-sink v0.3.4
   Compiling anyhow v1.0.27
   Compiling futures-task v0.3.4
   Compiling byteorder v1.3.4
   Compiling either v1.5.3
   Compiling pin-utils v0.1.0-alpha.4
   Compiling ppv-lite86 v0.2.6
   Compiling cc v1.0.50
   Compiling tower-layer v0.3.0
   Compiling itoa v0.4.5
   Compiling untrusted v0.7.0
   Compiling spin v0.5.2
   Compiling unicode-segmentation v1.6.0
   Compiling fixedbitset v0.2.0
   Compiling serde v1.0.104
   Compiling remove_dir_all v0.5.2
   Compiling multimap v0.8.0
   Compiling httparse v1.3.4
   Compiling ryu v1.0.3
   Compiling regex-syntax v0.6.17
   Compiling try-lock v0.2.2
   Compiling smallvec v1.2.0
   Compiling ansi_term v0.11.0
   Compiling percent-encoding v1.0.1
   Compiling futures-io v0.3.4
   Compiling tracing-core v0.1.10
   Compiling thread_local v1.0.1
   Compiling sharded-slab v0.0.8
   Compiling futures-channel v0.3.4
   Compiling futures-util v0.3.4
   Compiling itertools v0.8.2
   Compiling http v0.2.0
   Compiling heck v0.3.1
   Compiling indexmap v1.3.2
   Compiling num-traits v0.2.11
   Compiling num-integer v0.1.42
   Compiling aho-corasick v0.7.10
   Compiling want v0.3.0
   Compiling tracing-log v0.1.1
   Compiling base64 v0.10.1
   Compiling iovec v0.1.4
   Compiling net2 v0.2.33
   Compiling num_cpus v1.12.0
   Compiling which v3.1.1
   Compiling time v0.1.42
   Compiling quote v1.0.3
   Compiling ring v0.16.11
   Compiling rand_core v0.5.1
   Compiling petgraph v0.5.0
   Compiling mio v0.6.21
   Compiling prost-build v0.6.1
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.2.2
   Compiling rand v0.7.3
   Compiling chrono v0.4.11
   Compiling futures v0.3.4
   Compiling mio-uds v0.6.7
   Compiling http-body v0.3.1
   Compiling regex-automata v0.1.9
   Compiling regex v1.3.5
   Compiling tempfile v3.1.0
   Compiling matchers v0.0.1
   Compiling webpki v0.21.2
   Compiling sct v0.6.0
   Compiling rustls v0.16.0
   Compiling tokio-macros v0.2.5
   Compiling pin-project-internal v0.4.8
   Compiling prost-derive v0.6.1
   Compiling tracing-attributes v0.1.7
   Compiling serde_derive v1.0.104
   Compiling async-stream-impl v0.2.1
   Compiling async-trait v0.1.24
   Compiling tokio v0.2.13
   Compiling async-stream v0.2.1
   Compiling tracing v0.1.13
   Compiling pin-project v0.4.8
   Compiling tower-discover v0.3.0
   Compiling tower-util v0.3.0
   Compiling tower-load-shed v0.3.0
   Compiling tracing-futures v0.2.3
   Compiling prost v0.6.1
   Compiling prost-types v0.6.1
   Compiling tokio-util v0.2.0
   Compiling tower-load v0.3.0
   Compiling tower-timeout v0.3.0
   Compiling tower-make v0.3.0
   Compiling tower-retry v0.3.0
   Compiling tower-ready-cache v0.3.1
   Compiling tower-buffer v0.3.0
   Compiling tower-limit v0.3.0
   Compiling tokio-rustls v0.12.2
   Compiling tower-balance v0.3.0
   Compiling h2 v0.2.2
   Compiling tower v0.3.1
   Compiling tonic-build v0.1.1 (/Users/ddibiase/Projects/tonic/tonic-build)
   Compiling tonic-examples v0.1.0 (/Users/ddibiase/Projects/tonic/examples)
   Compiling hyper v0.13.3
error: failed to run custom build command for `tonic-examples v0.1.0 (/Users/ddibiase/Projects/tonic/examples)`

Caused by:
  process didn't exit successfully: `/Users/ddibiase/Projects/tonic/target/debug/build/tonic-examples-5fa24811858ec82b/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:1188:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by ddibiase** [comment]
I managed to get going with the example which depends on tonic 0.1, prost 0.6 and tonic-build 0.1. Seems like 0.1.1 has the breaking changes or something. :-/

**Comment by LucioFranco** [comment]
@ddibiase `data-prost` feature hasn't been released and is only active on master.

**Comment by ddibiase** [comment]
Thanks @LucioFranco I ended up figuring out what I did wrong and closed out the issue. You are correct it wasn't clear that data-prost wasn't fully supported. I'm now up and running with Tonic and its working beautifully. Thanks for your effort on this project. Also when you release the recent changes, would be extremely helpful if there was an in-depth migration or explanation of changes. I'll likely have to do this at some point and would truly appreciate the details. =)
